api/v2/api.md

```diff
@@ -0,0 +1,387 @@
+# dcrtime API Specification
+
+## V2
+
+This document describes the REST API provided by a `dcrtimed` server. This API 
+allows users to create and upload hashes which are periodically submitted to 
+the Decred blockchain. It gives the option to send a single string digest, as 
+well as multiples in a array of string digests. It also provides the ability 
+to confirm the addition of the hash to a timestamped collection along with 
+showing and validating their inclusion in the Decred blockchain.
+
+**Methods**
+
+- [`Timestamp`](#timestamp)
+- [`Timestamps`](#timestamps)
+- [`Verify`](#verify)
+
+**Return Codes**
+
+- [`ResultInvalid`](#ResultInvalid)
+- [`ResultOK`](#ResultOK)
+- [`ResultExistsError`](#ResultExistsError)
+- [`ResultDoesntExistsError`](#ResultDoesntExistsError)
+- [`ResultDisabled`](#ResultDisabled)
+
+### Methods
+
+#### `Timestamp`
+
+Upload one digest to the time server. The server will then add this digest to
+a collection and eventually to a transaction that goes in a Decred block. 
+This method returns immediately with the collection the digest has been added 
+to.You must use the verify call to find out when it has been anchored to a 
+block (which is done in batches at a set time interval that is not related 
+to the api calls).
+
+* **URL**
+
+  `/v2/timestamp/`
```

Comment:
Why do we need this route? We should force the use of `timestamps`.

I am ok if we make this a `GET` and provide the `timestamp` in the URL and add a `GET` verify as well that uses the URL for a single verify. With this, we can create a dumb NOJS tool that runs on v2.

@degeri what do you think?

Reply:
This route is needed for the nojs client version. It allows a client to send data with pure html, no javascript involved.

I'm working on the verify route and will commit it soon. I'll wait for @degeri 's input if we make it a `GET` or keep it a `POST`.

Reply:
There is no need to use GET we can do nojs with POST too. The data just cannot be JSON as I said above. The modifications done above for /v2/timestamp/ works with pure HTML.

Using GET might be easier to implement and has fewer potential pitfalls. We do have to take into account that all GET traffic is stored in the browser history (Not sure if thats super relevant). 

Would be good if we explicitly name ` /v2/timestamp/` as "nojs" so that people dont use it by mistake. For someone who looks at it for the first time /v2/timestamp/ seems simpler and easier to implement vs /v2/timestamps/ . We really should push most users towards `/v2/timestamp/` . Perhaps a big warning at doco is enough too.

Reply:
Actually I dont see why we cant support both GET and POST for `/v2/timestamp` user can decide based on their usage.

Reply:
Applied degeri's suggestions and the route now accepts both POST and GET requests.